Products/DataCollector/zenmodeler.py

```diff
@@ -1116,16 +1125,22 @@ def mainLoop(self, driver):
 
         # ZEN-26637 - did we collect during collector loop?
         self.didCollect = False
+        self.mainLoopGotDeviceList = False
         self.start = time.time()
+        self.collectorLoopIteration = self.collectorLoopIteration + 1
 
-        self.log.info("Starting collector loop...")
+        self.log.info("Starting collector loop #{:03d}...".format(self.collectorLoopIteration))
         yield self.getDeviceList()
         deviceList = driver.next()
         self.log.debug("getDeviceList returned %s devices", len(deviceList))
         self.log.debug("getDeviceList returned %s devices", deviceList)
         self.devicegen = iter(deviceList)
+        self.iterationDeviceCount = len(deviceList)
+        self.processedDevicesCount = 0
+        self.log.info("Got %d devices to be scanned during collector loop #%03d", self.iterationDeviceCount, self.collectorLoopIteration)
         d = drive(self.fillCollectionSlots)
         d.addErrback(self.fillError)
+        self.mainLoopGotDeviceList = True
```

Comment:
No, not necessarily, but I think in cases where it's not the case the conditionals on line [lineno.835](https://github.com/zenoss/zenoss-prodbin/pull/2154/files#diff-943de90b2eb194ce892ded9b2c846e2cR835) and/or [lineno.836](https://github.com/zenoss/zenoss-prodbin/pull/2154/files#diff-943de90b2eb194ce892ded9b2c846e2cR836) would be true and lineno.837 wouldn't be reached. I added initialization for the `mainGotDeviceList` instance variable, just in case.